### PR TITLE
Fix for the fact that windows file descriptors do not follow posix numbering standard

### PR DESCRIPTION
**Pull request by johnjuuljensen**
Resubmitting as requested.

Hey

So I tried to build and run under windows, but calling kept failing in fd_listen due to out of bounds check on the file descriptor.
I was pointed to this change: 49834b7
But thought it was somewhat invasive.

I ended up making this change instead. It's pretty simple and consists mostly of renaming of variables to clearly distinguish between what is an index into fhs and what is an actual descriptor handle.

The only new thing is the windows-only lookup_fd_index which maps file descriptors to indices.

Baresip compiles and runs and makes calls, when build with this version on windows SDK 10.

I have not tested anything under linux, not even compilation.

If you're interested, I can make any recommended changes and test it under linux.

Cheers,
John

### CODE REVIEW

**Comment by sreimers** [comment]
Can you fix the following code style errors:

```
The command "wget "https://raw.githubusercontent.com/baresip/baresip/master/test/ccheck.py"" exited with 0.
0.54s$ python3 ccheck.py
./src/main/main.c:227: C++ comment, use C comments /* ... */ instead
./src/main/main.c:227: line is too wide (83 - max 79)
./src/main/main.c:228: C++ comment, use C comments /* ... */ instead
./src/main/main.c:228: line is too wide (86 - max 79)
./src/main/main.c:237: C++ comment, use C comments /* ... */ instead
./src/main/main.c:596: C++ comment, use C comments /* ... */ instead
./src/main/main.c:598: C++ comment, use C comments /* ... */ instead
Statistics:
~~~~~~~~~~~
Number of files processed:     c: 202  cpp: 0  h: 69  mk: 49  m4: 0  py: 1  m: 0  s: 0  java: 0  php: 0 
Number of lines with errors:   7
```

File: src/main/main.c [hunk 1]
```diff
@@ -221,6 +223,26 @@ static struct re *re_get(void)
 
 #endif
 
+#ifdef WIN32
+// This code emulates possix numbering. There is no locking, so zero thread-safety.
```
Comment (sreimers) [inline src/main/main.c:227]: Wording s/possix/POSIX/

**Comment by sreimers** [comment]
Can you check again with "EPOLL", the patch runs in timeouts.

**Comment by johnjuuljensen** [comment]
That depends on whether epoll is supported on windows, because if not, I could try to run it under WSL, but I'd be at a disadvantage in debugging it there, since I haven't developed under linux in 20 years.

Mvh
John
________________________________
Fra: Sebastian Reimers <notifications@github.com>
sendt: lørdag den 19. september 2020 13.05
til: baresip/re
Cc: johnjuuljensen; Author
Emne: Re: [baresip/re] Fix for the fact that windows file descriptors do not follow posix numbering standard (#20)


Can you check again with "EPOLL", the patch runs in timeouts.

**Comment by sreimers** [comment]
Hi, I have fixed EPOLL and SELECT (`0` is a valid file descriptor). I close this pull request, can you check/test the new one?

https://github.com/baresip/re/pull/25

**Pull request closed without merging**